### PR TITLE
Context menu allows pasting any model bug fix

### PR DESCRIPTION
**Pull request by ric394**
resolves #8943

# Notes
- Extra checks have been added to prevent this bug from occurring.
- `Models.cs` was updated to allow models with no `[ValidParent()]` attributes to be handled properly.
- Tests updated to work with new functionality.
- `Nutrient`'s  had NutrientPatchManager added as valid parent.

### CODE REVIEW

File: Models/Core/ApsimFile/Structure.cs [hunk 1]
```diff
@@ -30,12 +30,13 @@ public static IModel Add(IModel modelToAdd, IModel parent)
...
-
-            // Call OnCreated
+            if(parent.IsChildAllowable(modelToAdd.GetType()))
+            {
+                parent.Children.Add(modelToAdd);
+            }
+            else throw new ArgumentException($"A {modelToAdd.GetType().Name} cannot be added to a {parent.GetType().Name}.");
+                       
             modelToAdd.OnCreated();
-            foreach (IModel model in modelToAdd.FindAllDescendants().ToList())
```
Comment (par456) [inline Models/Core/ApsimFile/Structure.cs:37]: Why did this get deleted? This is making that if a model creates children, that the children have their OnCreated called?

Comment (ric394) [reply Models/Core/ApsimFile/Structure.cs:37]: removed by mistaken. These lines have now been readded.

**Comment by ric394** [comment]
@hol353 this looks like POStats has run out of space. Are you able to assist?

**Comment by par456** [comment]
retest this please jenkins

**Pull request merged**